v3/src/components/graph/adornments/lsrl/lsrl-adornment-model.ts

```diff
@@ -0,0 +1,198 @@
+import { Instance, types } from "mobx-state-tree"
+import { Point } from "../../../data-display/data-display-types"
+import { AdornmentModel, IAdornmentModel, IUpdateCategoriesOptions, PointModel } from "../adornment-models"
+import { leastSquaresLinearRegression, tAt0975ForDf } from "../../utilities/graph-utils"
+import { kLSRLType } from "./lsrl-adornment-types"
+import { ICase } from "../../../../models/data/data-set-types"
+import { IGraphDataConfigurationModel } from "../../models/graph-data-configuration-model"
+import { ScaleNumericBaseType } from "../../../axis/axis-types"
+
+export const LSRLInstance = types.model("LSRLInstance", {
+  equationCoords: types.maybe(PointModel)
+})
+.volatile(() => ({
+  category: undefined as string | undefined,
+  intercept: undefined as number | undefined,
+  rSquared: undefined as number | undefined,
+  slope: undefined as number | undefined,
+  sdResiduals: undefined as number | undefined
+}))
+.views(self => ({
+  get isValid() {
+    return self.intercept && self.rSquared && self.slope && self.sdResiduals
```

Comment:
You probably want `isFinite()` here because this will fail when any of the values `=== 0`.